File: Dataset/Leetcode/valid/12/652.java
 class Solution {
    public String XXX(int num) {
        StringBuilder sb = new StringBuilder();
        int t = num / 1000;
        int h = (num / 100) % 10;
        int ty = (num / 10) % 10;
        int unit = num % 10;
        for (int i = 0; i < t; i++) {
            sb.append("M");
        }
        if (h == 9) {
            sb.append("CM");
        } else if (h == 4) {
            sb.append("CD");
        } else {
            if (h >= 5) {
                sb.append("D");
                h-=5;
            }
            for (int i = 0; i < h; i++) {
                sb.append("C");
                
            }
        }
        if (ty == 9) {
            sb.append("XC");
        } else if (ty == 4) {
            sb.append("XL");
        } else {
            if (ty >= 5) {
                sb.append("L");
                ty-=5;
            }
            for (int i = 0; i < ty; i++) {
                sb.append("X");

            }
        }
        if (unit == 9) {
            sb.append("IX");
        } else if (unit == 4) {
            sb.append("IV");
        } else {
            if (unit >= 5) {
                sb.append("V");
                unit-=5;
            }
            for (int i = 0; i < unit; i++) {
                sb.append("I");

            }
        }
        return new String(sb);
    }
}

File: Dataset/Leetcode/train/19/652.cpp
 class Solution {
public:
    ListNode* XXX(ListNode* head, int n) {
        ListNode* prep[30];
        int sum = 0;
        while (head != nullptr) {
            prep[sum++] = head;
            head = head->next;
        }
        if (sum == n) {  return prep[sum-n]->next; }
        else if (n == 1){
            prep[sum-1-n]->next = nullptr;
        } else {
            prep[sum-1-n]->next = prep[sum+1-n];
        }
        return prep[0];
    }
};

